### PR TITLE
Features/upload image

### PR DESCRIPTION
**Pull request by jrief**
Added ``ImageField`` which together with ng-file-upload can be used to handle form submissions via JSON/Ajax.

### CODE REVIEW

**Comment by jrief** [comment]
@adrienbrunet @jkosir 
would you please be so kind and review this pull request.
It adds support for dragging and dropping images into forms which can be submitted via Ajax.

This pull request will trigger version 0.9

File: djng/forms/fields.py [hunk 1]
```diff
@@ -14,3 +25,85 @@ def widget_attrs(self, widget):
...
+                if value['temp_name'] is False and value['previous_image']:
+                    # Delete previous image
+                    previous_image = self.signer.unsign(value['previous_image'])
+                    self.remove_images(previous_image)
+                    return False
+                if value['temp_name'] is None or value['temp_name'] is True:
+                    # Nothing changed
+                    return
+            except (KeyError, TypeError, signing.BadSignature) as excp:
+                raise ValidationError("Got bogous upstream data")
```
Comment (adrienbrunet) [inline djng/forms/fields.py]: "bogous" --> "bogus" ?

File: docs/upload-image.rst [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+	    $httpProvider.defaults.headers.common['X-CSRFToken'] = '{{ csrf_token }}';
+	}]);
+	</script>
+
+
+Caveats
+=======
+
+When users upload images, but never submit the corresponding form, the folder holding these
+temporary images gets filled up. Therefore you should some kind of (cron)job which cleans up that
```
Comment (adrienbrunet) [inline docs/upload-image.rst]: "you should *add*"

File: examples/server/tests/test_fileupload.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    def test_receive_small_image(self):
+        content = self.upload_image()
+        content['file:0'].pop('url')
+        data = {'avatar': content['file:0']}
+        form = TestUploadForm(data=data)
+        self.assertTrue(form.is_valid())
+        self.assertIsInstance(form.cleaned_data['avatar'], InMemoryUploadedFile)
+        self.assertEquals(form.cleaned_data['avatar'].name, "sample-image.jpg")
+
+        # TODO: delete this image again
```
Comment (adrienbrunet) [inline examples/server/tests/test_fileupload.py]: Do you need help to end this or we should get rid off this?

Comment (jrief) [inline examples/server/tests/test_fileupload.py]: I have to create a model to test it. That's something I was trying to avoid.

**Comment by rfleschenberg** [comment]
Will this work with alternative storage backends?

**Comment by jrief** [comment]
superseded by PR #303 -> closing

**Pull request closed without merging**